Apps/Src/ReadCarCAN.c

```diff
@@ -189,53 +147,75 @@ void Task_ReadCarCAN(void *p_arg)
     );
     assertOSError(OS_READ_CAN_LOC, err);
 
-    //Start CAN Watchdog timer
+    // Start CAN Watchdog timer
     OSTmrStart(&canWatchTimer, &err);
     assertOSError(OS_READ_CAN_LOC, err);
-    
 
-    while (1)
-    {
+    #ifndef __TEST_READCARCAN
+    Minion_Error_t Merr; // Not used when testing since ignition won't be changed
+    #endif
+
+    while(1){
+
+        #ifndef __TEST_READCARCAN
+        arrayIgnitionStatus = Minion_Read_Pin(IGN_1, &Merr); // Don't check ignition if just testing
+        #endif
+
+        // Array Contactor is turned off if Ignition 1 is off
+        if(arrayIgnitionStatus == false){
+            Contactors_Set(ARRAY_CONTACTOR, OFF, true);
+
+            // Update Display
+            UpdateDisplay_SetArray(false);
+        }
         
-        //Get any message that BPS Sent us
-        ErrorStatus status = CANbus_Read(&dataBuf,true,CARCAN);  
-        if(status != SUCCESS) {
+        // BPS sent a message
+        ErrorStatus status = CANbus_Read(&dataBuf, true, CARCAN);
+        if(status != SUCCESS){
             continue;
         }
```

Comment:
I don't think this does anything

Reply:
@NathanielDelgado Got it. It looks like whenever there isn't a success, the ID is set to null so it wouldn't enter the switch case. I'll go ahead and remove it, unless there is something else to consider.

Reply:
I deleted it, but it's being caught as an unused variable. Could it be helpful in the long run to keep it?

Reply:
Nevermind, keep it. It's better to have that case be more apparent @diyarajon